### PR TITLE
Add TLS support to the pinniped-proxy directly.

### PR DESCRIPTION
**Pull request by absoludity**
Signed-off-by: Michael Nelson <minelson@vmware.com>

### Description of the change

After playing around with different options, this appears to be the simplest way to add tls support to pinniped-proxy without breaking the existing non-tls setup.

I'm leaving this as a draft until I figure out how TLS on the related service will interact with this. After that, it should just be updating the api server to be able to trust the cert, then the related chart changes.

### Applicable issues

- fixes #

### CODE REVIEW

File: cmd/pinniped-proxy/src/main.rs [hunk 1]
```diff
@@ -28,34 +32,58 @@ async fn main() -> Result<()> {
...
-            }))
-        }
-    });
-
     let addr = ([0, 0, 0, 0], opt.port).into();
 
-    let server = Server::bind(&addr).serve(make_svc);
-
-    info!("Listening on http://{}", addr);
+    let with_tls = opt.proxy_tls_cert != "" && opt.proxy_tls_cert_key != "";
```
Comment (antgamdia) [inline cmd/pinniped-proxy/src/main.rs:37]: Maybe worth adding a warning if just one of them is set ?

**Review by antgamdia — APPROVED** [review]
Great! Thank for the explanation!

File: cmd/pinniped-proxy/src/main.rs [hunk 2]
```diff
@@ -28,34 +32,58 @@ async fn main() -> Result<()> {
...
 
     // Run the server for ever. If it returns with an error, return the
     // result, otherwise, if it completes, we return Ok.
-    server.await?;
+    if with_tls {
+        info!("Configuring with TLS cert {} and key {}", opt.proxy_tls_cert, opt.proxy_tls_cert_key);
```
[the remark this answers is not in the document]
Comment (absoludity) [reply cmd/pinniped-proxy/src/main.rs:42]: Nope, they're just file paths. Not so clear from the option names...

File: cmd/pinniped-proxy/src/main.rs (hunk 1 above)
Comment (absoludity) [reply cmd/pinniped-proxy/src/main.rs:37]: Even erroring and stopping - we can't run with just one.

**Pull request merged**